### DISCUSSION TITLE
250 change limit on Azure Pipeline

### DISCUSSION THREAD

**Discussion by james1301** [rfennell/AzurePipelines#1871] [General]
Hi I was wondering why I wasn't getting all of the related work items on my pipeline to produce the release notes and came across this: https://stackoverflow.com/questions/72491130/azure-devops-incomplete-list-of-related-workitems-in-build-pipeline. Which seems that the pipeline is limited to 250 commits.

Does anybody know about this? Or how to get around it? Messes up what would be a nice solution otherwise.

**Comment by rfennell** [comment]
My release notes task should get up to 5000 items (what I think this the upper limit for batch size)

` await (buildApi.getBuildChanges((currentBuildArtifact as any).properties.projectId, (currentBuildArtifact as any).versionId, "", 5000)),`

This is just a call to 

[`GET https://dev.azure.com/{organization}/{project}/_apis/build/builds/{buildId}/changes?api-version=7.1-preview.2`](https://learn.microsoft.com/en-us/rest/api/azure/devops/build/builds/get-build-changes?view=azure-devops-rest-7.1) if you want to test it in a browser

**Comment by james1301** [comment]
This only seems to return 50 items, which seems a very strange round number.

How does the 5000 get specified?

**Reply by rfennell** [reply]
if yo using the REST API you set the `$top` option, 5000 is the maximum

**Reply by james1301** [reply]
Ahh yeah sorry, I think I skimmed over that parameter name 🤦‍♂️.

**Reply by rfennell** [reply]
It is not the most obviously named one, I had a read it a couple of time to remember :)

**Reply by james1301** [reply]
I only seem to ever get back 200 when I change the $top parameter. I can change it down and it goes down, but is 200 some limit?

**Comment by rfennell** [comment]
Quiet probably, I am accessing via the Node SDK wrapper, so I assume it setting it doing the paging behind the scenes and the has a max 500 for performance

**Reply by james1301** [reply]
Sorry what do you mean? Do you mean 200* page size?

And in your code, you do repeated calls to other pages?

**Reply by rfennell** [reply]
Any paging code is within the Node SDK, I just call

` await (buildApi.getBuildChanges((currentBuildArtifact as any).properties.projectId, (currentBuildArtifact as any).versionId, "", 5000)),`
